WordPress/wordcamp.org
# Tools: Update JS dependencies

Update JS dependencies to latest versions across all workspaces (projects). Keeping these up to date ensures any tests better reflect production, and can bring in improvements in build steps, etc.

The relevant changes here are:

- The eslint rules were updated slightly, so there are changes in how jsdoc is formatted. These were mostly whitespace that was auto-fixed.
- Updating to the latest `@wordpress/scripts` means we can remove our custom CSS build config, simplifying `blocks/webpack.config.js`.
- Due to how the default command works, all blocks `style.scss` files had to be renamed to `edit.scss`, otherwise they are pulled into a second CSS file.
- Built CSS files are not named `*.min.css` anymore, so the enqueuing logic was updated
- The `wp-scripts` command won't take entries on the CLI in the same format anymore, so projects with custom entrypoints now have basic `webpack.config.js` files. This is more understandable than the CLI approach, and still inherits all the other default config from wp-scripts's config.

The format command was also updated, but note that we can't easily use it without changing the base `.eslintrc.js` to use `plugin:@wordpress/eslint-plugin/recommended`, which would change formatting in a way that I don't think we all agree on.

### How to test the changes in this Pull Request:

1. Tests here should pass
2. Build all the files, `yarn workspaces run build`
3. Spot-check that there are no functional changes
